Source/NSubstitute/SubstituteExtensions.cs

```diff
@@ -142,7 +142,7 @@ internal static ConfiguredCall Returns<T>(MatchArgs matchArgs, T returnThis, par
             }
             else
             {
-                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese));
+                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese).ToArray());
```

Comment:
I think that since ReturnMultipleValues is backed by ConcurrentQueue it should not accept IEnumerable as an argument. You can create an IEnumerable instance that produces infinite number of values.